src/components/RestaurantList/RestaurantList.jsx

```diff
@@ -30,47 +32,53 @@ const RestaurantList = ({ categories }) => {
     setSortList(actionOrder[caseOrder]);
   }
   // logic for the filters 
-  const filteredData = () =>{
+  const filteredData = () => {
     let result = [];
-
+  
     const categories = searchParams.getAll('cuisine');
     const rating = searchParams.get('rating');
+  
     if (searchParams.get('searchTerm')) {
-      result = data.filter((element) => element.restaurantName.replaceAll(' ', '').toLowerCase().includes(searchParams.get('searchTerm')))
+      result = data.filter((element) =>
+        element.restaurantName
+          .replaceAll(' ', '')
+          .toLowerCase()
+          .includes(searchParams.get('searchTerm'))
+      );
     }
-
-    const filterByCategorie = (arr) =>{
-      return arr.filter((element1)=>{
-        return element1.categories.some((element2)=> {
-          return categories.includes(element2)
-        })
-      })
+  
+    const filterByCategorie = (arr) => {
+      return arr.filter((element1) => {
+        return element1.categories.some((element2) => {
+          return categories.includes(element2);
+        });
+      });
     };
-    if (rating >= 2 && categories.length === 0) result = data.filter(element =>  Math.round(element.rating) >= rating);
-    if (categories.length > 0 && rating < 2)  result = filterByCategorie(data)
-    if (categories.length > 0 && rating >= 2) result = data.filter(element => Math.round(element.rating) >= rating && filterByCategorie(data));
-    return result
+  
+    if (rating >= 2 && categories.length === 0)
+      result = data.filter((element) => Math.round(element.rating) >= rating);
+    if (categories.length > 0 && rating < 2) result = filterByCategorie(data);
+    if (categories.length > 0 && rating >= 2)
+      result = data.filter(
+        (element) =>
+          Math.round(element.rating) >= rating && filterByCategorie([element]).length > 0
```

Comment:
This should be:
```
!filterByCategory([element]).length
```

Reply:
Changed the name, but with the negation operator it never returns the array empty so it can never the display the message that no restaurants match that search